[Security Solution][Data Quality Dashboard] fix tests and potential tests timing out on ci

kapral18 opened the pull request:
addresses #196216

Removing accessibility selectors to ensure 10x speed of tests with data-test-subj locators.

<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

kapral18 committed:
[Security Solution][Data Quality Dashboard] fix tests and potential tests timing out on ci

addresses #196216

elasticmachine commented:
Pinging @elastic/security-threat-hunting-explore (Team:Threat Hunting:Explore)

elasticmachine commented:
Pinging @elastic/security-threat-hunting (Team:Threat Hunting)

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/243255)
* Commit: 0b199daddcf6bcb821f3f4f800a84456c79ed2f2

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243255","number":243255,"commit":"0b199daddcf6bcb821f3f4f800a84456c79ed2f2"}],"number":243255}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

PhilippeOberti reviewed (approved):
Thanks! Maybe that something we should make at a higher level? We've seen some issues on the @elastic/security-threat-hunting-investigations side as well for the more complex tests we have!

kapral18 committed:
improve(test): Remove accessibility selectors in slow tests

- Refactored test cases to use `data-test-subj` attributes for 10x faster element selection
- Added missing `data-test-subj` attributes in components to facilitate the prev point

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/247431)
* Commit: 9f63b9b20cd50330f7abd58bfdb11b6e2427c71a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/6ef1bced47082280e0ca2afb239c78b23c070005) | [after](https://github.com/elastic/kibana/commit/9f63b9b20cd50330f7abd58bfdb11b6e2427c71a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 21.0MB | 21.0MB | +651.0B |



### History
* :green_heart: [Build #243255](https://buildkite.com/elastic/kibana-pull-request/builds/243255) succeeded 0b199daddcf6bcb821f3f4f800a84456c79ed2f2

cc @kapral18

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/247431","number":247431,"commit":"9f63b9b20cd50330f7abd58bfdb11b6e2427c71a"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/243255","number":243255,"commit":"0b199daddcf6bcb821f3f4f800a84456c79ed2f2"}],"number":247431}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11634800031

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/198717">](https://github.com/elastic/kibana/pull/198717)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->

kapral18 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.16|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/200549">](https://github.com/elastic/kibana/pull/200549)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.8 -->
